catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/util/EndpointUtilTest.java

```diff
@@ -0,0 +1,204 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.opengis.filter.Filter;
+import org.opengis.filter.Or;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.AttributeRegistry;
+import ddf.catalog.data.InjectableAttribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.AttributeBuilder;
+import ddf.catalog.filter.ContextualExpressionBuilder;
+import ddf.catalog.filter.EqualityExpressionBuilder;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+public class EndpointUtilTest {
+
+    @InjectMocks
+    EndpointUtil endpointUtil;
+
+    List<MetacardType> metacardTypeList;
+
+    MetacardType metacardTypeMock;
+
+    CatalogFramework catalogFrameworkMock;
+
+    FilterBuilder filterBuilderMock;
+
+    List<InjectableAttribute> injectableAttributeList;
+
+    InjectableAttribute injectableAttributeMock;
+
+    AttributeRegistry attributeRegistryMock;
+
+    Filter filterMock;
+
+    AttributeBuilder attributeBuilderMock;
+
+    ContextualExpressionBuilder contextualExpressionBuilderMock;
+
+    QueryResponse responseMock;
+
+    Metacard metacardMock;
+
+    Result resultMock;
+
+    @Before
+    public void setUp() throws Exception {
+
+        // mocks
+        metacardTypeList = new ArrayList<>();
+        metacardTypeMock = mock(MetacardType.class);
+        metacardTypeList.add(metacardTypeMock);
+
+        catalogFrameworkMock = mock(CatalogFramework.class);
```

Comment:
✏️ Instead of mocking the `CatalogFramework` and all the calls the `ResultIterator` makes to it, we could mock the `ResultIterator` directly using a `BiFunction<CatalogFramework, QueryRequest, ResultIterable>`, or a factory method (see https://confluence.di2e.net/display/DDT/Dependency+Injection, Injecting Non-Singleton Dependencies).